plugins/gcp-projects/src/components/ProjectDetailsPage/ProjectDetailsPage.tsx

```diff
@@ -60,19 +60,13 @@ const DetailsPage = () => {
   const api = useApi(gcpApiRef);
   const classes = useStyles();
 
-  const {
-    loading,
-    error,
-    value: details,
-  } = useAsync(
-    async () =>
-      api.getProject(
-        decodeURIComponent(location.search.split('projectId=')[1]),
-      ),
-    [location.search],
+  const [{ status, result: details, error }, { execute }] = useAsync(async () =>
+    api.getProject(decodeURIComponent(location.search.split('projectId=')[1])),
   );
 
-  if (loading) {
+  useMountEffect(execute);
```

Comment:
Although gotta say, this looks really tedious

Reply:
agree. also the state === 'loading' meh.